### ISSUE TITLE
mongodb: crashes on start

### ISSUE THREAD

**Issue by jsumners** [voidlinux/void-packages#4106]
I have a fresh install of Void (installed on 2016-04-14). I have not been able to get mongodb to launch on it. I just tried again after the 3.2.5_1 package was released. It is still crashing on launch:

```
2016-04-15T20:19:46.708-0400 I CONTROL  [main] ***** SERVER RESTARTED *****
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] MongoDB starting : pid=10846 port=27017 dbpath=/var/lib/mongodb 64-bit host=ice-guild.info
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] db version v3.2.4
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] git version: e2ee9ffcf9f5a94fad76802e28cc978718bb7a30
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] OpenSSL version: LibreSSL 2.3.3
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] allocator: tcmalloc
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] modules: none
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] build environment:
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten]     distarch: x86_64
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten]     target_arch: x86_64
2016-04-15T20:19:46.719-0400 I CONTROL  [initandlisten] options: { config: "/etc/mongodb/mongodb.conf", net: { bindIp: "127.0.0.1" }, storage: { dbPath: "/var/lib/mongodb" }, systemLog: { destination: "file", logAppend: true, path: "/var/log/mongodb/mongod.log", quiet: true } }
2016-04-15T20:19:46.770-0400 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=1G,session_max=20000,eviction=(threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),checkpoint=(wait=60,log_size=2GB),statistics_log=(wait=0),
2016-04-15T20:19:46.792-0400 E STORAGE  [initandlisten] WiredTiger (22) [1460765986:792619][10846:0x7fe791097cc0], connection: unknown compressor 'snappy': Invalid argument
2016-04-15T20:19:46.796-0400 I -        [initandlisten] Fatal Assertion 28561
2016-04-15T20:19:46.796-0400 I -        [initandlisten]

***aborting after fassert() failure
```

This configuration being used:

```yaml
systemLog:
  quiet: true
  destination: file
  path: /var/log/mongodb/mongod.log
  logAppend: true
  logRotate: reopen

storage:
  dbPath: /var/lib/mongodb
  journal:
    enabled: true
  wiredTiger:
    engineConfig:
      journalCompressor: none

processManagement:
  fork: false

# network interfaces
net:
  port: 27017
  bindIp: 127.0.0.1

security:
  authorization: disabled
```

**Comment by xtraeme** [comment]
Try to see if https://github.com/voidlinux/void-packages/commit/c92b2adf237cbf80d95de7e4b31807ed20f1a17f helps. Wait until it's built or build it manually to test.

**Comment by jsumners** [comment]
@xtraeme Sorry, no change. I disabled the `wiredTiger` config in my configuration so that it wold use the defaults and it complained:

```
WiredTiger (22) [1462022359:635875][3757:0x7f382c522cc0], connection: unknown compressor 'snappy': Invalid argument
```

So I re-enabled my configuration to disable compression and got the same crash as I reported originally.

I am clearing out `/var/lib/mongodb` before starting the process. So it's not corrupt data.

**Comment by jsumners** [comment]
I just updated to:

+ mongodb-3.2.6_3
+ wiredtiger-2.8.0_4

Tried the same configurations as in my last report and still got the same errors.

**Comment by Vaelatern** [comment]
Which platform are you on?

**Comment by Vaelatern** [comment]
Ah, `x86_64-musl`, I should read more carefully.

**Comment by jsumners** [comment]
No. Just plain x86_64. 

![image](https://cloud.githubusercontent.com/assets/321201/15111209/878cf540-15b3-11e6-8384-d1226e610521.png)

**Comment by jsumners** [comment]
I just downloaded the live ISO, installed it in a VM, and then installed mongodb -- https://www.youtube.com/watch?v=n8HXzLTnfyQ

Same result.

**Comment by WKuipers** [comment]
I've had this bug a few versions back, was able to fix it by

- completely removing the package 
- reinstalling 
- Modify the configuration to not use any compression
- Run

This is of course a workaround

**Comment by jsumners** [comment]
@WKuipers just tried it. Same result.

**Comment by WKuipers** [comment]
@jsumners did you also wipe the database?

**Comment by jsumners** [comment]
@WKuipers yes, every time.

**Comment by WKuipers** [comment]
@jsumners this `/etc/mongodb/mongodb.cong `

**Comment by jsumners** [comment]
@WKuipers turning off the block compressor did it. Thanks, I didn't know that was a thing.

**Comment by antonio-malcolm** [comment]
Has anyone raised this issue to the folks at MongoDB or WiredTiger?

Not sure it helps any with diagnosis, but I also encountered this issue, after compiling from source, over multiple attempts, with different configuration options (so, I'm not certain the issue is Void-specific- mayhps I'll attempt it on OS X, or another distro, later).

I used zlib, snappy, pcre, and pkg-config, et al. (there were others, already installed, for other server packages), from the XBPS packages. I compiled Boost from source, as I needed a newer version, as it's shared with other server packages (namely, MariaDB).

I had to disable both the journal compression and the collection block compression, to start the server:

```
storage:
  wiredTiger:
    collectionConfig:
      blockCompressor:
        none
    engineConfig:
      journalCompressor:
        none
```

**Comment by antonio-malcolm** [comment]
BTW, I notice, per the XBPS template, for MongoDB, you're still attempting to buiild the tools with scons.
Does that actually work? AFAIK, the tools are now in their own project, have been, for some time, and are written in Go.

(Moving tools required for the most basic DB backup functionality/redundancy into another project, in another language, is one of several reasons I am quickly growing tired of MongoDB.)

**Comment by stevelitt** [comment]
Danger Will Robinson!!! The mongod executable has no default config file location!!! If you want a config file, it must be specified as -f <location> or --config <location>. Void Linux' mongod runit run script specifies the config at /etc/mongodb/mongodb.conf and supplies a config file there. So you add to /etc/mongodb/mongodb.conf:

- wiredTigerJournalCompressor=none
- wiredTigerCollectionBlockCompressor=none

The important thing to remember is that if you're running mongod from a  command prompt, you MUST specify a config file argument in order to have a config file. The executable has no default config file.

**Comment by stevelitt** [comment]
Also, please remember Mongodb supports two styles of config files:

- The .ini style like old MSDOS .ini files, which is what Void gives you in /etc/mongodb/mongodb.conf
- The newer YAML style. View `https://docs.mongodb.com/v3.0/reference/configuration-options/`  for details on the YAML style and other great information.

**Comment by jsumners** [comment]
@stevelitt I think everyone involved in the conversation knows those things.

**Comment by ebfe** [comment]
Still an issue?

**Comment by jsumners** [comment]
I haven't noticed it. But I also haven't removed the configuration that got it working.

**Comment by infowolfe** [comment]
Still an issue on default install x86_64.

**Comment by AandersonL** [comment]
i am facing the same issue in default install X86_64

**Comment by linde12** [comment]
I'm still facing the same issue as the OP and would love some help. I'm using mongodb-3.4.9_5

**Comment by linde12** [comment]
@Vaelatern Exactly. It works if i do as @WKuipers suggested and start mongo with `mongod --config /path/to/cfg`

**Comment by Yaroslav-95** [comment]
This issue still exists, @WKuipers 's workaround still works.
